tests/unit/models/search.js

```diff
@@ -0,0 +1,287 @@
+        dois: ["10.1136/bmj.h2575"]
+      }
+    };
+
+    expect(search.shouldEnhance(scope)).toEqual(true);
+  });
+
+  it("should not enhance a journal search result without an issn or eissn", function() {
+    var scope = {
+      document: {
+        content_type: "Journal"
+      }
+    };
+
+    expect(search.shouldEnhance(scope)).toEqual(false);
+  });
+
+  it("should not enhance an article search result without a doi", function() {
+    var scope = {
+      document: {
+        content_type: "Journal Article"
+      }
+    };
+
+    expect(search.shouldEnhance(scope)).toEqual(false);
+  });
+
+  it("should not enhance a journal search result without a content type", function() {
+    var scope = {
+      document: {
+        issns: ["0028-4793"]
+      }
+    };
+
+    expect(search.shouldEnhance(scope)).toEqual(false);
+  });
+
+  it("should not enhance an article search result without a content type", function() {
+    var scope = {
+      document: {
+        dois: ["10.1136/bmj.h2575"]
+      }
+    };
+
+    expect(search.shouldEnhance(scope)).toEqual(false);
+  });
+
+  it("should build a journal endpoint for a journal search result", function() {
+    var scope = {
+      document: {
+        content_type: "Journal",
+        issns: ["0028-4793"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("search?issns=00284793");
+  });
+
+  it("should select the issn over the eissn when a journal search result includes both", function() {
+    var scope = {
+      document: {
+        content_type: "Journal",
+        issns: ["0028-4793"],
+        eissns: ["0082-3974"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("search?issns=00284793");
+  });
+
+  it("should select the eissn when the journal search result has no issn", function() {
+    var scope = {
+      document: {
+        content_type: "Journal",
+        eissns: ["0082-3974"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("search?issns=00823974");
+  });
+
+  it("should build an article endpoint for an article search result", function() {
+    var scope = {
+      document: {
+        content_type: "Journal Article",
+        dois: ["10.1136/bmj.h2575"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("articles/doi/10.1136%2Fbmj.h2575");
+  });
+
+  it("should build an article endpoint for an article search result and include its journal", function() {
+    var scope = {
+      document: {
+        content_type: "Journal Article",
+        dois: ["10.1136/bmj.h2575"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("articles/doi/10.1136%2Fbmj.h2575?include=journal");
+  });
+
+  it("should include a browzineWebLink in the BrowZine API response for a journal", function() {
+    var data = search.getData(journalResponse);
+    expect(data).toBeDefined();
+    expect(search.getBrowZineWebLink(data)).toEqual("https://browzine.com/libraries/XXX/journals/10292");
```

Comment:
I'd move strings like these that are shared with strings set earlier into test constants.

Reply:
Wrestled with this too, but prioritized test independence over reuse.